### PR TITLE
Fixes #20491: Error message in technique editor doesn't help

### PR DESCRIPTION
**Pull request by ElaadF**
https://issues.rudder.io/issues/20491
Examples:
image
image

### CODE REVIEW

**Comment by ElaadF** [comment]
Commit modified

File: webapp/sources/rudder/rudder-web/src/main/elm/build-app.sh [hunk 1]
```diff
@@ -21,6 +21,6 @@ PROJECTS=("Notifications" "Healthcheck" "Editor" "Onboarding" "Rules")
 for PROJECT in ${PROJECTS[*]}; do
   lower=$(echo "${PROJECT}" | tr '[:upper:]' '[:lower:]')
   cd ${ELM_DIR}/${lower}
-  elm-0.19.1 make --optimize sources/${PROJECT}.elm --output=generated/rudder-${lower}.js
+  elm-0.19.1 make  sources/${PROJECT}.elm --output=generated/rudder-${lower}.js
```
Comment (amousset) [inline webapp/sources/rudder/rudder-web/src/main/elm/build-app.sh:24]: is this change voluntary?

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/cfclerk/domain/VariableAndSectionSpec.scala [hunk 2]
```diff
@@ -407,8 +407,8 @@ object SectionVariableSpec {
       case SELECT1 => SelectOneVariableSpec(varName, description, longDescription,
         valueslabels, multivalued, checked, constraint)
       case REPORT_KEYS =>
-        if(providedValues.isEmpty) throw EmptyReportKeysValue(varName)
-        else PredefinedValuesVariableSpec(varName, description, (providedValues.head, providedValues.tail),
+        if(test.isEmpty) throw EmptyReportKeysValue(varName)
```
Comment (amousset) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/cfclerk/domain/VariableAndSectionSpec.scala:410]: looks like test values

**Comment by ElaadF** [comment]
PR updated with a new commit

File: webapp/sources/rudder/rudder-web/src/main/elm/build-app.sh (hunk 1 above)
Comment (ElaadF) [reply webapp/sources/rudder/rudder-web/src/main/elm/build-app.sh]: not at all, pushed by mistake (but `--optimize` prevent from logging into browser console, not convenient for debugging...)

Comment (amousset) [reply webapp/sources/rudder/rudder-web/src/main/elm/build-app.sh]: isn't this the point of the build-dev.sh script?

Comment (ElaadF) [reply webapp/sources/rudder/rudder-web/src/main/elm/build-app.sh]: The `build-dev.sh` script call `build-app.sh` to compile elm files and then move compilated file into the right directory, but with `--optimize` option the compilation fails if we use logging lib in Elm files, so everytime we have to remove this option temporarily, but I often forget to put it back before push my changes (in fact I used `rudder-dev wip` and then `rudder-dev commit` so it pushed all changed files in my repo)

**Review by fanf — APPROVED** [review]

**Comment by ElaadF** [comment]
OK, squash merging this PR

**Pull request merged**